pyanaconda/modules/payloads/source/source_base.py

```diff
@@ -134,3 +136,31 @@ def tear_down_with_tasks(self):
         """
         task = TearDownMountTask(self._mount_point)
         return [task]
+
+
+class RPMSourceInterface(ABC):
+    """Interface class which has to be implented by all sources used by DNF payload."""
+
+    @abstractmethod
+    def generate_repo_configuration(self):
+        """Generate RepoConfigurationData structure.
+
+        This structure will be used by DNF payload in the main process.
+
+        FIXME: This is a temporary solution. Will be removed after DNF payload logic is moved.
+        """
+        pass
+
+
+class RPMMountingSourceBase(MountingSourceBase, ABC):
+    """Enhanced MountingSourceBase class to handle code specific for RPM sources."""
+    # pylint: disable=abstract-method
+
+    def generate_repo_configuration(self):
+        """Generate RepoConfigurationData structure."""
+        data = RepoConfigurationData()
+
+        data.name = self.type.value
+        data.url = join_paths("file:///", self.mount_point)
```

Comment:
I would prefer to see the implementation in every RPM source. Maybe, the class `RepoConfigurationData` could have a class method `for_file` that would accept a name and a path and a return a new instance of the data object.